src/lazy/binary/system/lazy_sequence.rs

```diff
@@ -0,0 +1,252 @@
+use crate::element::{Annotations, Element, IntoAnnotatedElement, Sequence, Value};
+use crate::lazy::binary::raw::lazy_raw_sequence::{LazyRawSequence, RawSequenceIterator};
+use crate::lazy::binary::system::lazy_value::AnnotationsIterator;
+use crate::lazy::binary::system::lazy_value::LazyValue;
+use crate::{IonError, IonResult, IonType, SymbolTable};
+use std::fmt;
+use std::fmt::{Debug, Formatter};
+
+/// A list or S-expression in a binary Ion stream whose header has been parsed but whose body
+/// (i.e. its child values) have not. A `LazySequence` is immutable; its data can be read any
+/// number of times.
+///
+/// ```
+///# use ion_rs::IonResult;
+///# fn main() -> IonResult<()> {
+///
+/// // Construct an Element and serialize it as binary Ion.
+/// use ion_rs::element::Element;
+/// use ion_rs::ion_list;
+/// use ion_rs::lazy::binary::lazy_reader::LazyReader;
+///
+/// let element: Element = ion_list! [10, 20, 30].into();
+/// let binary_ion = element.to_binary()?;
+///
+/// let mut lazy_reader = LazyReader::new(&binary_ion)?;
+///
+/// // Get the first value from the stream and confirm that it's a list.
+/// let lazy_list = lazy_reader.expect_next()?.read()?.expect_list()?;
+///
+/// // Visit the values in the list
+/// let mut sum = 0;
+/// for lazy_value in &lazy_list {
+///     // Read each lazy value in the lazy list as an int (i64) and
+///     // add it to the running total
+///     sum += lazy_value?.read()?.expect_i64()?;
+/// }
+///
+/// assert_eq!(sum, 60);
+///
+/// // Note that we can re-read any of the lazy values. Here we'll step into the list again and
+/// // read the first child value.
+/// let first_int = lazy_list.iter().next().unwrap()?.read()?.expect_i64()?;
+/// assert_eq!(first_int, 10);
+///
+///# Ok(())
+///# }
+/// ```
+pub struct LazySequence<'top, 'data> {
+    pub(crate) raw_sequence: LazyRawSequence<'data>,
+    pub(crate) symbol_table: &'top SymbolTable,
+}
```

Comment:
The only reason they don't is because I didn't need it to get this far. 😛 I'm fine with adding it later; I imagine that will be useful when we go to implement `LazyElement`.